majit/majit-ir/src/box_ref.rs

```
@@ -0,0 +1,1093 @@
+        self.write_forwarded(Forwarded::None);
+    }
+
+    /// Slice 8.D writer: route to `op.forwarded` / `inputarg.forwarded`
+    /// when the box is bound, else fall back to the in-Box slot. The
+    /// Box.forwarded slot is inert for bound boxes — `get_forwarded`
+    /// already reads from the same handle that gets written here, so a
+    /// redundant Box write would only desync if a future reader looked at
+    /// the wrong slot. `op_handle` and `inputarg_handle` are mutually
+    /// exclusive (BoxKind determines which is populated by bind_*).
+    fn write_forwarded(&self, value: Forwarded) {
+        if let Some(weak) = self.0.op_handle.borrow().as_ref() {
```

Clippy (1.95.0)

[warning] this `if` statement can be collapsed

(warning)